File: crates/klyron_deploy/bindings/cpp/builder.cpp
#include "builder.hpp"

namespace klyron::_deploy {

DeployBuilder::DeployBuilder() : config_{} {}

DeployBuilder& DeployBuilder::set_config(const DeployConfig& cfg) {
    config_ = cfg;
    return *this;
}

DeployConfig DeployBuilder::config() const {
    return config_;
}

}

File: crates/klyron_deploy/bindings/rust/builder.rs
//! Builder for klyron_deploy
use crate::types::DeployConfig;

pub struct DeployBuilder {
    config: DeployConfig,
}

impl DeployBuilder {
    pub fn new() -> Self {
        Self { config: DeployConfig::default() }
    }
    pub fn with_version(mut self, version: &str) -> Self {
        self.config.version = version.to_string();
        self
    }
    pub fn build(self) -> DeployConfig {
        self.config
    }
}
